File: python/209.minimum-size-subarray-sum.py
class Solution:
    def solve(self, target, nums):
        l=0
        res = float('inf')
        current_sum = 0
        for i in range(len(nums)):
            current_sum += nums[i]
            while l <= i and current_sum >= target:
                res = min(res, i-l+1)
                current_sum -= nums[l]
                l +=1

        if res == float('inf') :
            return 0
        return res

    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        return self.solve(target, nums)
        

File: cpp/209.minimum-size-subarray-sum.cpp
class Solution {
public:
    int solve(int target, vector<int> &nums){
	int l = 0, n = nums.size(), res = INT_MAX;
	int current_sum = 0;
	for(int i = 0 ; i < n ;i++){
	   current_sum += nums[i];
	   while(current_sum >= target && l <= i){
		   res = min(res, i-l+1);
		   current_sum -= nums[l++];
	   }
	}
	if(res == INT_MAX) return 0;
	return res;
    }
    int minSubArrayLen(int target, vector<int>& nums) {
	return solve(target, nums);
    }
};
/*
 target = 7, nums = [2,3,1,2,4,3]
2
     2  3  1  2  4  3
                 l
                    r
 * */
